### PR TITLE
mpsl: fem: initial support for nRF2220 and nRF2240 on nRF54L

### PR DESCRIPTION
**Pull request by ankuns**
This PR adds initial support for the nRF2220 and nRF2240 FEMs for nRF54L devices.
Agreed with PMT that no doc is provided yet.
In the dts bindings the default output power for nrf2240 is set to 14dBm.

The nRF2220EK and nRF2240EK shields are given support for `nrf54l15dk/nrf54l15/cpuapp` board.
Please note that this DK is not arduino compatible and requires separate handling within shield code.

Minor fixes to fem_al related to nRF54L are provided.

Details in commit messages.

### CODE REVIEW

File: boards/shields/nrf2220ek/boards/nrf54l15dk_nrf54l15_cpuapp.overlay
Comment (e-rk) [inline boards/shields/nrf2220ek/boards/nrf54l15dk_nrf54l15_cpuapp.overlay:89]: These overrides are not needed and they don't do anything.

File: boards/shields/nrf2240ek/boards/nrf54l15dk_nrf54l15_cpuapp.overlay
Comment (e-rk) [inline boards/shields/nrf2240ek/boards/nrf54l15dk_nrf54l15_cpuapp.overlay:116]: These overrides are not needed and they don't do anything.

File: boards/shields/nrf2240ek/boards/nrf54l15dk_nrf54l15_cpuapp.conf
Comment (e-rk) [inline boards/shields/nrf2240ek/boards/nrf54l15dk_nrf54l15_cpuapp.conf:7]: Since `DPPIC10` is used directly with the nrfx_dppi API, I suggest putting `CONFIG_NRFX_DPPI10=y` here. `CONFIG_NRFX_PPIB11=y` and `CONFIG_NRFX_PPIB21=y` as well.
`CONFIG_NRFX_DPPI0=y` should be set as well on nRF53.
Since GPPI is also used, `CONFIG_NRFX_GPPI` can be added as well.

**Review by e-rk — CHANGES_REQUESTED** [review]

File: boards/shields/nrf2240ek/boards/nrf54l15dk_nrf54l15_cpuapp.overlay
Comment (ankuns) [reply boards/shields/nrf2240ek/boards/nrf54l15dk_nrf54l15_cpuapp.overlay:116]: In fact `CONFIG_HAS_HW_NRF_DPPIC` is defined only when we have `status = "okay";` in a dppicXX.
By default the status is "disabled".

File: boards/shields/nrf2220ek/boards/nrf54l15dk_nrf54l15_cpuapp.overlay
Comment (ankuns) [reply boards/shields/nrf2220ek/boards/nrf54l15dk_nrf54l15_cpuapp.overlay:89]: lets continute in comments below https://github.com/nrfconnect/sdk-nrf/pull/19124#discussion_r1861950864 , as they are about the same issue, just for nrf2220 not nrf2240.

File: boards/shields/nrf2240ek/boards/nrf54l15dk_nrf54l15_cpuapp.overlay
Comment (ankuns) [reply boards/shields/nrf2240ek/boards/nrf54l15dk_nrf54l15_cpuapp.overlay:116]: We agreed they are necessary. You turn them on globally in PR https://github.com/zephyrproject-rtos/zephyr/pull/82254 which must later go to sdk-zephyr, this will take time. Would you agree on taking a technical debt here? The non-necessary overrides will be removed from here later when the change from Zephyr propagates downstream.

File: subsys/mpsl/fem/nrf2220/mpsl_fem_nrf2220.c
Comment (piotrkoziar) [inline subsys/mpsl/fem/nrf2220/mpsl_fem_nrf2220.c:118]: Can we have an explanation of the hardware connections for nRF2220 in the form of a comment section? Something like [this comments for simple gpio](https://projecttools.nordicsemi.no/bitbucket/projects/DRGN/repos/dragoon/browse/mpsl/libs/fem/src/simple_gpio/mpsl_fem_simple_gpio_dppi.c?at=3f1cacfd80a9db13bea3530327188262a560c410#17).

I have an analogic comment for nRF2240.

**Review by piotrkoziar — APPROVED** [review]

File: subsys/mpsl/fem/nrf2220/mpsl_fem_nrf2220.c
Comment (ankuns) [reply subsys/mpsl/fem/nrf2220/mpsl_fem_nrf2220.c:118]: @piotrkoziar  such doc is provided within the repo you pointed, just in other file.

**Review by e-rk — APPROVED** [review]

**Comment by ankuns** [comment]
@nrfconnect/ncs-co-build-system @nrfconnect/ncs-co-drivers  @nrfconnect/ncs-si-muffin 
please review.

**Review by nordicjm — APPROVED** [review]

**Review by maje-emb — APPROVED** [review]
fem_al looks good.

**Review by anangl — APPROVED** [review]

**Comment by ankuns** [comment]
@martintv could you review this PR please, you're a code owner. This PR can't be merged without your approval.

**Review by martintv — APPROVED** [review]

**Pull request merged**
